### ISSUE TITLE
Unable to re-iterate Muxes

### ISSUE THREAD

**Issue by ejhumphrey** [pescadores/pescador#86]
After a day of struggles (and a night of bad pescadreams), I do believe I've identified the source of one of the issues I've been having.

**tl;dr:** When a mux expires *or* is interrupted, it can't return a fresh iterator for subsequent iteration.

This behavior is captured with [this test](https://github.com/pescadores/pescador/blob/ejh_20170403_test_refactor/tests/test_mux.py#L252), which produces the following mismatched output:
```
['a', 'd', 'e...'c', 'f', ...] == []
```
Here, the mux works once (left side), but can't return a fresh iterator to reproduce its sequence.

As for *why* this is happening ... not 100% sure yet, but I think it's because there's some iterator-dependent state getting set in `__init__` instead of `activate`, such as the [source stream distribution](https://github.com/pescadores/pescador/blob/ejh_20170403_test_refactor/pescador/mux.py#L152). May be other things, but won't be clear until digging in.

@bmcfee any thoughts on this? @cjacoby and I discussed, and the plan is to fix on a separate PR after #80 is fully back.
